Provide `--summary` flag to generate the license summary file (#103)
apache/skywalking-eyes: README.md

This command serves as assistance for human beings to audit the dependencies license, it's exit code is always 0.

We also support two flags:

|Flag name|Short name|Description|
|---------|----------|-----------|
|`--output`|`-o`|Save the dependencies' `LICENSE` files to a specified directory so that you can put them in distribution package if needed.|
|`--summary`|`-s`|Based on the template, aggregate all dependency information and generate a `LICENSE` file.|

```bash

> license-eye -c test/testdata/.licenserc_for_test_check.yaml dep resolve -o ./dependencies/licenses -s LICENSE.tpl

Reviewer: Can you also add an section to demonstrate how the `LICENSE.tpl` looks like and what variables it can use?

Author: Updated.

</details>

##### Summary Template

The summary is a template to generate the summary of dependencies' licenses based on the [Golang Template](https://pkg.go.dev/text/template). It includes these variables:

|Name|Type|Example|Description|
|----|----|-------|-----------|

> |LicenseContent|string|`{{.LicenseContent}}`|The project license content, it's relate to the `header.license.spdx-id` config. |

Reviewer: ```suggestion
|LicenseContent|string|`{{.LicenseContent}}`|The project license content, it's the license of `header.license.spdx-id` (if set), otherwise it's the `header.license.content`. |
```

> |Groups|list structure|`{{ range .Groups }}`|The dependency group, all license is aggregate by the same license [SPDX ID](https://spdx.org/licenses/). |

Reviewer: ```suggestion
|Groups|list structure|`{{ range .Groups }}`|The dependency groups, all licenses are grouped by the same license [SPDX ID](https://spdx.org/licenses/). |
```

|Groups.Name|string|`{{.Name}}`|The [SPDX ID](https://spdx.org/licenses/) of dependency. |

> |Groups.Deps|list structure|`{{ range .Deps }}`|All dependencies, it including all of same [SPDX ID](https://spdx.org/licenses/) dependencies under `Groups.Name`. |

Reviewer: ```suggestion
|Groups.Deps|list structure|`{{ range .Deps }}`|All dependencies with the same [SPDX ID](https://spdx.org/licenses/). |
```

|Groups.Deps.Name|string|`{{.Name}}`|The name of the dependency. |
|Groups.Deps.Version|string|`{{.Version}}`|The version of the dependency. |
|Groups.Deps.LicenseID|string|`{{.LicenseID}}`|The [SPDX ID](https://spdx.org/licenses/) of the dependency license. |

<details>
<summary>Summary Template Generate</summary>

Summary template content:
```
{{.LicenseContent }}
{{ range .Groups }}
========================================================================

> {{.Name}} licenses

Reviewer: Do you think it'd be more specific if we rename `{{.Name}}` to something like `{{.LicenseSpdxID}}` or `{{.LicenseName}}`? Because there is another `{{.Name}}` inside the `{{.Deps}}`,

Author: I think it could be better to rename it to `{{.LicenseID}}`? WDYT? 
Let it same with the `{{.LicenseID}}` inside the `{{.Deps}}`. For now, they are the same.

Reviewer: > I think it could be better to rename it to `{{.LicenseID}}`? WDYT?

`{{.LicenseID}}` is good to me

Author: Updated.
